src/View/Model/ResourceViewModel.php

```diff
@@ -19,45 +19,62 @@
 namespace ZfrRest\View\Model;
 
 use Zend\View\Model\ViewModel;
-use ZfrRest\Resource\ResourceInterface;
 
 /**
  * @author  Michaël Gallego <mic.gallego@gmail.com>
  * @licence MIT
  */
-class ResourceModel extends ViewModel
+class ResourceViewModel extends ViewModel
 {
     /**
-     * ResourceModel is usually terminate
-     *
      * @var bool
      */
     protected $terminate = true;
 
     /**
-     * The resource to be rendered
-     *
-     * @var ResourceInterface
+     * @var string|null
+     */
+    protected $version = null;
+
+    /**
+     * @param array $variables
+     * @param array $options
      */
-    protected $resource;
+    public function __construct(array $variables = [], array $options = [])
+    {
+        $this->setVariables($variables, true);
+
+        foreach ($options as $key => $value) {
+            switch ($key) {
+                case 'version':
+                    $this->version = (string) $value;
+                    break;
+
+                case 'template':
+                    $this->template = (string) $options['template'];
+                    break;
+            }
+        }
+    }
 
     /**
-     * Constructor
+     * Explicitly set a version
      *
-     * @param ResourceInterface $resource
+     * @param  string $version
+     * @return void
      */
-    public function __construct(ResourceInterface $resource)
+    public function setVersion($version)
```

Comment:
Yep, because you may decide not to set a specific version, but then the plugin that resolve it to a version needs to set the resolved version to the resource view model (so that it's correctly propagated to any children resources).